third_party/move/move-vm/runtime/src/loader/mod.rs

```diff
@@ -1742,21 +1742,28 @@ impl Loader {
                 .into_iter()
                 .unzip();
 
-        // For aggregators / snapshots, the first field should be lifted.
-        if let Some(kind) = &maybe_mapping {
-            if let Some(l) = field_layouts.first_mut() {
-                *l = MoveTypeLayout::Tagged(
-                    LayoutTag::IdentifierMapping(kind.clone()),
-                    Box::new(l.clone()),
-                );
-            }
-        }
-
         let has_identifier_mappings =
             maybe_mapping.is_some() || field_has_identifier_mappings.into_iter().any(|b| b);
 
         let field_node_count = *count - count_before;
-        let struct_layout = MoveStructLayout::new(field_layouts);
+        let layout = if Some(IdentifierMappingKind::DerivedString) == maybe_mapping {
+            // For DerivedString, the whole object should be lifted.
+            MoveTypeLayout::Tagged(
+                LayoutTag::IdentifierMapping(IdentifierMappingKind::DerivedString),
+                Box::new(MoveTypeLayout::Struct(MoveStructLayout::new(field_layouts))),
```

Comment:
I think it depends what you do with ValueImpl::DelayedField, but I think for Aggregators you need to have max outside of it, (because of the width and correctness on changes and stuff), so not sure if that is possible/better

Reply:
Ithink single field has the same size, but good point